library/core/src/option.rs

```diff
@@ -985,6 +986,83 @@ impl<T> Option<T> {
     }
 }
 
+impl<T> Option<&T> {
+    /// Converts from `Option<&T>` (or `&Option<&T>`) to `*const T`.
```

Comment:
>  (or `&Option<&T>`)

Maybe this should take an `Option<&T>` by value instead? Or I suppose that might be problematic with the `&mut` versions?

Regardless, it's probably better to leave out the ``(or `&Option<&T>`)`` part from the documentation, as that just might add confusion.

Reply:
I added that for consistency with the docs of [`Option::as_deref`](https://doc.rust-lang.org/stable/std/option/enum.Option.html#method.as_deref) and [`Option::as_deref_mut`](https://doc.rust-lang.org/stable/std/option/enum.Option.html#method.as_deref_mut). I agree it doesn't add much value.